File: app/src/main/java/com/dtp/chamber/database/AndroidDatabaseManager.java
package com.dtp.chamber.database;

import android.database.sqlite.SQLiteOpenHelper;

import com.dtp.Database;
import com.dtp.DatabaseManager;

/**
 * Created by ner on 4/6/17.
 */

public class AndroidDatabaseManager implements DatabaseManager {

    private SQLiteOpenHelper openHelper;

    public AndroidDatabaseManager(SQLiteOpenHelper openHelper) {
        this.openHelper = openHelper;
    }

    @Override
    public Database beginTransaction() {
        return new AndroidDatabase(openHelper.getWritableDatabase());
    }

    @Override
    public boolean endTransaction() {
        openHelper.close();

        return true;
    }
}

File: app/src/main/java/com/dtp/chamber/database/AndroidDatabaseManager.kt
package com.dtp.chamber.database

import android.database.sqlite.SQLiteOpenHelper
import com.dtp.DataStoreIn

import com.dtp.Database
import com.dtp.DatabaseManager

/**
 * Created by ner on 4/6/17.
 */

class AndroidDatabaseManager(private val openHelper: SQLiteOpenHelper) : DatabaseManager {

    override fun beginTransaction(): Database = AndroidDatabase(openHelper.writableDatabase)

    override fun endTransaction(): Boolean {
        openHelper.close()

        return true
    }

    override fun createInDataStore(): DataStoreIn = AndroidDataStoreIn()
}
